crates/duty-tracker/src/contract_state_machine.rs

```diff
@@ -754,14 +874,45 @@ impl ContractSM {
     fn process_graph_nonces(
         &mut self,
         signer: P2POperatorPubKey,
+        claim_txid: Txid,
         nonces: Vec<PubNonce>,
     ) -> Result<Option<OperatorDuty>, TransitionErr> {
         match &mut self.state.state {
-            ContractState::Requested { graph_nonces, .. } => {
-                graph_nonces.insert(signer, nonces);
+            ContractState::Requested {
+                peg_out_graphs,
+                graph_nonces,
+                ..
+            } => {
+                let unpacked = PogMusigF::unpack(nonces).ok_or(TransitionErr(
+                    "could not unpack nonce vector into PogMusigF".to_string(),
+                ))?;
+                let Some(session_nonces) = graph_nonces.get_mut(&claim_txid) else {
+                    return Err(TransitionErr(format!(
+                        "could not process graph nonces. claim_txid ({}) not found in nonce map",
+                        claim_txid
+                    )));
+                };
+                session_nonces.insert(signer, unpacked);
                 Ok(
-                    if graph_nonces.len() == self.cfg.operator_table.cardinality() {
-                        Some(OperatorDuty::PublishGraphSignatures)
+                    if session_nonces.len() == self.cfg.operator_table.cardinality() {
+                        let Some((input, _)) = peg_out_graphs.get(&claim_txid) else {
+                            return Err(TransitionErr(format!(
+                                "could not process graph nonces. claim_txid ({}) not found in peg out graph map" ,
+                                claim_txid
```

Comment:
I think you can do this for better readability:
```suggestion
                                "could not process graph nonces. claim_txid {claim_txid} not found in peg out graph map"
```